Branch-SDK/src/main/java/io/branch/referral/validators/BranchIntegrationModel.java

```diff
@@ -63,6 +64,17 @@ private void updateDeepLinkSchemes(Context context) {
         }
     }
 
+    //Function to get value from string resource
+    public String getStringResource(String stringWithIdValue){
+        if(stringWithIdValue.contains("resourceID")){
+            String resourceIdArr[] = stringWithIdValue.split(" ");
+            if(context!=null && resourceIdArr!=null && resourceIdArr.length==2){
+                return context.getResources().getString(Integer.parseInt(resourceIdArr[1].replace("0x",""),16));
```

Comment:
I know this is just a validator but this code looks very fragile, e.g. is it always base 16 and contains "0x"? If so, perhaps you can leave a comment saying so and/or giving examples?

Better yet, you could add a couple try/catches and rethrow the exception with the appropriate exception message (e.g. "could not read the string resource X").